### PR TITLE
OCTOPUS-625:Added role for approve and issue csr cert and get vm file

### PR DESCRIPTION
**Pull request by pkenchap**
Added the role for approve and issue csr certificate . 
Also added get vm file with wait until VM comes to active state.

### CODE REVIEW

File: openstack/intel-worker/playbooks/roles/approve_and_issue/files/approve_and_issue.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  fi
+  IDX=$(($IDX + 1))
+
+  # Wait for 30 seconds before we hammer the system
+  echo "Sleeping before re-running - 30 seconds"
+  sleep 30
+
+  # Re-read the 'Ready' count
+  READY_COUNT=$(oc get nodes -l kubernetes.io/arch=amd64 | grep "${MACHINE_PREFIX}" | grep -v NotReady | grep -c Ready)
+done
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/approve_and_issue/files/approve_and_issue.sh:82]: Hey @pkenchap @Chandan-Abhyankar refactored this script recently at https://github.com/IBM/ocp4-upi-compute-powervs/blob/main/ansible/post/files/approve_and_issue.sh. it'd be worth pulling that in here. and removing the reference to the proxy.

**Review by prb112 — CHANGES_REQUESTED** [review]
a couple of suggestions are left inline

File: openstack/intel-worker/playbooks/roles/approve_and_issue/files/approve_and_issue.sh (hunk 1 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/approve_and_issue/files/approve_and_issue.sh]: @prb112 , Used the same script and modified with removing the proxy .

File: openstack/intel-worker/playbooks/intel-worker-playbook.yml [hunk 2]
```diff
@@ -14,6 +14,18 @@
   roles:
     - virtual_machine_create
 
+- name: Wait for 3 minutes
+  hosts: localhost
+  tasks:
+    - name: Pause for 3 minutes
+      pause:
+        seconds: 180
```
[the remark this answers is not in the document]
Comment (pkenchap) [reply openstack/intel-worker/playbooks/intel-worker-playbook.yml]: Yes you are right , it's not required now. removed it.

**Comment by pkenchap** [comment]
Hi Paul,

Looping I updated the script and it works seemlessly for the approve and issue .
Here I had created a 3 VM's and all approved and added to the Cluster
`
        "Approve and Issue - #3",

        "List of Intel Workers to be added with prefix 'rdr-mac-worker-openstack': ",
        "rdr-mac-worker-openstack-1   NotReady                   worker   2m6s    v1.28.3+4cbdd29",
        "rdr-mac-worker-openstack-2   NotReady                   worker   2m3s    v1.28.3+4cbdd29",
        "rdr-mac-worker-openstack-3   Ready                      worker   59s     v1.28.3+4cbdd29",
        "",
        "Approving: csr-c79xz system:node:rdr-mac-worker-openstack-1",
        "certificatesigningrequest.certificates.k8s.io/csr-c79xz approved",
        "Approving: csr-s4sks system:node:rdr-mac-worker-openstack-1",
        "certificatesigningrequest.certificates.k8s.io/csr-s4sks approved",
        "Approving: csr-tjrnc system:node:rdr-mac-worker-openstack-1",
        "certificatesigningrequest.certificates.k8s.io/csr-tjrnc approved",
        "Approving: csr-wlshr system:node:rdr-mac-worker-openstack-1",
        "certificatesigningrequest.certificates.k8s.io/csr-wlshr approved",
        "Approving: csr-z2c89 system:node:rdr-mac-worker-openstack-1",
        "certificatesigningrequest.certificates.k8s.io/csr-z2c89 approved",
        "Approving: csr-zkbl5 system:node:rdr-mac-worker-openstack-1",
        "certificatesigningrequest.certificates.k8s.io/csr-zkbl5 approved",
        "Approving: csr-f7m8z system:node:rdr-mac-worker-openstack-2",
        "certificatesigningrequest.certificates.k8s.io/csr-f7m8z approved",
        "Approving: csr-lcnkp system:node:rdr-mac-worker-openstack-2",
        "certificatesigningrequest.certificates.k8s.io/csr-lcnkp approved",
        "Approving: csr-m6khm system:node:rdr-mac-worker-openstack-2",
        "certificatesigningrequest.certificates.k8s.io/csr-m6khm approved",
        "Approving: csr-rb42m system:node:rdr-mac-worker-openstack-2",
        "certificatesigningrequest.certificates.k8s.io/csr-rb42m approved",
        "Approving: csr-wdrvh system:node:rdr-mac-worker-openstack-2",
        "certificatesigningrequest.certificates.k8s.io/csr-wdrvh approved",
        "Approving: csr-zqwqc system:node:rdr-mac-worker-openstack-2",
        "certificatesigningrequest.certificates.k8s.io/csr-zqwqc approved",
        "Sleeping before re-running - 30 seconds",
        "Supplied Worker/s with prefix: 'rdr-mac-worker-openstack' are Ready.",
        "rdr-mac-worker-openstack-1   Ready                      worker   3m9s    v1.28.3+4cbdd29",
        "rdr-mac-worker-openstack-2   Ready                      worker   3m6s    v1.28.3+4cbdd29",
        "rdr-mac-worker-openstack-3   Ready                      worker   2m2s    v1.28.3+4cbdd29"
`

I had made the looping in the playbook will go in iteration of creating the VM's as per the user requirement of worker nodes.

image

image

**Review by prb112 — APPROVED** [review]

**Pull request merged**